independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AbstractGenerator.java

```diff
@@ -42,7 +42,7 @@ static String generatedNameFromTarget(String targetPackage, String baseName, Str
         if (targetPackage == null || targetPackage.isEmpty()) {
             return baseName + suffix;
         } else {
-            return targetPackage.replace('.', '/') + "/" + baseName + suffix;
+            return (targetPackage.isEmpty() ? "" : targetPackage.replace('.', '/') + "/") + baseName + suffix;
```

Comment:
This can never happen, see the branch above - for `targetPackage.isEmpty()` we always return `baseName + suffix`...

Reply:
ack - fixed.